classes/evasys_synchronizer.php

```diff
@@ -65,15 +68,19 @@ public function get_evasys_courseid() {
             $extras = [];
         }
         // If noone has associated the course itself, we force that.
-        if (!in_array($maincourse, $extras)) {
-            array_unshift($extras, $maincourse);
+        if (isset($maincourse)) {
+            if (!in_array($maincourse, $extras) && !$maincourse == "") {
```

Comment:
!= will do the same type conversion. I believe you're looking for !==

Reply:
The problem is, that in `!$maincourse == ""`, even if both `$maincourse` and `""` are strings, `$maincourse` will still be converted to a boolean, so that the not can be applied. That leads to this behaviour:
`!"asdf" == "asdf"`   ~>   `!true == "asdf"`   ~>   `false == "asdf"` ~> `false`
`!"asdf" == "jklö"`   ~>   `!true == "jklö"`   ~>   `false == "jklö"` ~> `false`
It should theoretically work in this case, because you are comparing with an empty string, but I believe it would still be better to use either != or !== (or `!($maincourse == "")` ) to prevent confusion

Reply:
Yes, Justus is correct. `!$x` should never be followed by a comparison operator, because that will rarely do what you expect it to do. 

In this case, `!empty($maincourse)` should also do what is expected. Changed that.